hutch_python/utils.py

```diff
@@ -433,3 +436,22 @@ def get_fully_qualified_domain_name():
         )
         logger.debug("getfqdn failed", exc_info=True)
         return ""
+
+
+abort_msg = """
+Your RunEngine has been interrupted.  If you aborted the current run (Ctrl+C),
+the RunEngine is ready for use.  If you paused the run, you must resume,
+abort, stop, halt the RunEngine before continuing.
+"""
+
+
+class AbortSigintHandler(SignalHandler):
+    def __init__(self, RE):
+        super().__init__(signal.SIGINT, log=RE.log)
+        self.RE = RE
+
+    def handle_signals(self):
+        # Check for pause requests from keyboard.
+        if self.RE.state.is_running and (not self.RE._interrupted):
```

Comment:
Interrupted status seems like a thing that external code should be query without poking around with RE internals.

I think that might be a quick bluesky PR to add such a property, if you're up for it...

Reply:
Is the interrupted check necessary here? What does it do?

Reply:
Ah that may be vestigial.  The old RE could be in an interrupted and running state if a plan was paused.  I think the way we have it now, that shouldn't occur (we interrupt and immediately reset the RE).  If we find a way to capture a different signal for pausing the plan, this might be necessary though

Reply:
Looks like this became relevant again?

Reply:
> Looks like this became relevant again?

Justifying extraneous documentation the hard way